src/internal/text/TextBody.tsx

```diff
@@ -1,13 +1,12 @@
-import type { ReactNode } from 'react';
+import { TextAs } from './TextAs';
+import type { TextReact } from './types';
 
-type TextBodyReact = {
-  children: ReactNode;
-};
-
-export function TextBody({ children }: TextBodyReact) {
+export function TextBody(props: TextReact) {
   return (
-    <span className="text-base text-black text-sans leading-normal">
-      {children}
-    </span>
+    <TextAs
+      as="span"
```

Comment:
For example, this I consider a step backwards in internal DevX. Every frontend developer immediately knows what a `span` is, but it takes me clicking in to the `TextAs` component, reading it, and zooming out to understand this custom syntax. It also takes more keystrokes to express the same thing ("this is a span").

Reply:
@ilikesymmetry I feel the same, but let me reflect very quickly on what's going on here.

I think here the goal is to optimize the internal use of TextBody, not to expose `TextAs`.

Internal components are a bit more about effeciency, and in some cases sacrificies usability. Which means do more with less.

The real question here to ask is, are we ever going to use `as="span"` as feature? like have `h3` or `h4` in it? So far we don't. But this could be easily be work that helps us get there if need it.

So if we are thinking to use `as` feature relativelly soon, that it makes sense keep it. If not, than we should just use `span`. 

@kyhyco what's your answear here.

Reply:
Here are the alternative:
```javascript
export function TextBody(props: TextReact) {
  const color = useTextColor(color);
  const className = cn('font-sans text-base leading-normal', color);

   // class assignment behavior from next PR
   const color = useTextColor(color);
   const className= cn('font-sans text-base leading-normal', color, props.className);

    // no text color assignment
    const className=cn('font-sans text-base leading-normal', props.className);

  return (
    <span className={className} {...props} />
  )
}
```

The primary reason for `TextAs` is to consolidate the color assignment logic across multiple text components without having to copy paste it every time.`TextAs` is a super simple component as well. 

Here are the benefits of this whole approach:
- For potentially addressing accessibility, it would be nice to address that in a single line PR by exposing `as` on `TextReact` type.

- For DX, when I look at a UX design, I look for the text name + color. I would love for the autocompletion to kick in for selecting the component and color.

- For maintainability, you only need to modify `TextAs` and `TextReact` without copy pasting new business logic. We also guarantee that share logic will be in `TextAs`.

- For adding new Text component, I think this is neat:
  - copy paste any regular text component
  - change name
  - set default tag
  - update tailwind styles

Yes, I agree `span` is much nicer to read on initial reading. But you don't get the benefits of centralized logic in a single file. That is also nice.

Reply:
Great context, thank you @kyhyco 
Ok so based on that it makes sense to keep this implementation.